File: 7570.js
const fs = require('fs');

function solve() {
	const input = fs.readFileSync('/dev/stdin').toString().split('\n');
	let inputIdx = 0;
	const N = +input[inputIdx++];
	const arr = input[inputIdx++].split(' ').map((v) => +v);
	const len = [];
	len.length = N + 1;
	for (let i = 0; i < N + 1; len[i++] = 0);

	for (const num of arr) {
		len[num] = len[num - 1] + 1;
	}

	return N - Math.max(...len);
}

console.log(solve());

File: 7570.py
from sys import stdin

input = stdin.readline


def solve():
    N = int(input())
    arr = [*map(int, input().split())]
    length = [0] * (N + 1)

    for num in arr:
        length[num] = length[num - 1] + 1

    return N - max(length)


if __name__ == '__main__':
    print(solve())

